milestone2.ts

```diff
@@ -125,37 +158,44 @@ let kyle: Volunteer = {
 /* Lets make a clone of Kyle above using the spread operator and assign it to kyleClone
  */
 type CopyVolunteer = (vol: Volunteer) => Volunteer;
-let copyVolunteer: CopyVolunteer; // Code here
+let copyVolunteer: CopyVolunteer = (vol) => {
+  const volunteerClone: Volunteer = {...vol}
+  return volunteerClone;
+}
 
 // let kyleClone: Volunteer = copyVolunteer(kyle);
-//console.log(kyleClone);
+// console.log(kyleClone);
 
 // Question 6: Spread Operator Part 2
 /* Next, lets use the spread operator to update your kyleClone object with the updatedLocation defined below and assign it to kyleNew
  */
 let updatedLocation = { city: "Seattle", state: "WA" };
-let kyleNew: Volunteer;
 type UpdateVolunteer = (
   vol: Volunteer,
   updates: Partial<Volunteer>
 ) => Volunteer;
-let updateVolunteer: UpdateVolunteer; // Code here
+let updateVolunteer: UpdateVolunteer = (vol, updatedLocation) => {
+  const kyleNew: Volunteer = {...vol, ...updatedLocation}
+  return kyleNew
+};
 
-// let kyleNew = updateVolunteer(kyleClone);
-//console.log(kyleNew);
+// let kyleNew = updateVolunteer(kyleClone, updatedLocation);
+// console.log(kyleNew);
 
 // Question 7: Object Destructuring
 /* Now that we have our updated kyle, lets use object destructuring to get his name, age, and city
  */
 
 type GetVolunteerInfo = (vol: Volunteer) => String;
 let getVolunteerInfo: GetVolunteerInfo = (vol) => {
-  let name, age, city; // Code here
+  let name = vol.name;
+  let age = vol.age;
+  let city = vol.city;
```

Comment:
Please use [object destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#object_destructuring) for this question